packages/@aws-cdk/custom-resources/lib/aws-custom-resource/runtime/index.ts

```diff
@@ -86,7 +86,20 @@ export async function handler(event: AWSLambda.CloudFormationCustomResourceEvent
     console.log(JSON.stringify(event));
     console.log('AWS SDK VERSION: ' + AWS.VERSION);
 
-    let physicalResourceId = (event as any).PhysicalResourceId;
+    // Default physical resource id
+    let physicalResourceId: string;
+    switch (event.RequestType) {
+      case 'Create':
+        physicalResourceId = event.ResourceProperties.Create?.physicalResourceId ??
+                             event.ResourceProperties.Update?.physicalResourceId ??
+                             event.ResourceProperties.Delete?.physicalResourceId ??
+                             event.LogicalResourceId;
```

Comment:
Another option here is to make `physicalResourceId` mandatory also for the Delete call...

Reply:
I am good with this.